### ISSUE TITLE
New IPNS key types

### ISSUE THREAD

**Issue by ianopolous** [libp2p/specs#589]
We're hoping to use new post-quantum (or hybrid) key types in IPNS in the next few years in Peergos and wanted to start a discussion about how to add new key types. In particular, ones that have 8k-17k signatures. The latter of which would put them over the 10k IPNS record limit. 

The bigger issue is that you can't start using a new key type until most clients have upgraded to support it. 

NB: I'm not talking about peerIds here if they can be decoupled.

**Comment by ianopolous** [comment]
Closing in favour of https://github.com/ipfs/specs/issues/448
